src/Collections/MemoryConfigCollection.php

```diff
@@ -0,0 +1,306 @@
+        return $result;
+    }
+
+    public function exists($class, $name = null, $options = 0)
+    {
+        $config = $this->get($class, null, $options);
+        if (!isset($config)) {
+            return false;
+        }
+        if ($name) {
+            return array_key_exists($name, $config);
+        }
+        return true;
+    }
+
+    public function remove($class, $name = null)
+    {
+        $class = strtolower($class);
+        if ($name) {
+            unset($this->config[$class][$name]);
+        } else {
+            unset($this->config[$class]);
+        }
+        // Discard call cache
+        unset($this->callCache[$class]);
+        return $this;
+    }
+
+    public function removeAll()
+    {
+        $this->config = [];
+        $this->metadata = [];
+        $this->history = [];
+        $this->callCache = [];
+    }
+
+    /**
+     * Get complete config (excludes middleware-applied config)
+     *
+     * @return array
+     */
+    public function getAll()
+    {
+        return $this->config;
+    }
+
+    /**
+     * @deprecated 4.0...5.0
+     *
+     * Synonym for merge()
+     *
+     * @param string $class
+     * @param string $name
+     * @param mixed  $value
+     * @return $this
+     */
+    public function update($class, $name, $value)
+    {
+        $this->merge($class, $name, $value);
+        return $this;
+    }
+
+    public function merge($class, $name, $value)
+    {
+        // Detect mergeable config
+        $existing = $this->get($class, $name, true);
+        if (is_array($value) && is_array($existing)) {
+            $value = Priority::mergeArray($value, $existing);
+        }
+
+        // Apply
+        $this->set($class, $name, $value);
+        return $this;
+    }
+
+    public function getMetadata()
+    {
+        if (!$this->trackMetadata || !is_array($this->metadata)) {
+            return [];
+        }
+
+        return $this->metadata;
+    }
+
+    public function getHistory()
+    {
+        if (!$this->trackMetadata || !is_array($this->history)) {
+            return [];
+        }
+
+        return $this->history;
+    }
+
+    public function serialize()
+    {
+        return serialize(
+            [
+            $this->config,
+            $this->history,
+            $this->metadata,
+            $this->trackMetadata,
+            $this->middlewares,
+            ]
+        );
+    }
+
+    public function unserialize($serialized)
+    {
+        list(
+            $this->config,
+            $this->history,
+            $this->metadata,
+            $this->trackMetadata,
+            $this->middlewares
+        ) = $this->unserialize($serialized);
+    }
+
+    public function nest()
+    {
+        return clone $this;
+    }
+
+    /**
+     * Save metadata for the given class
+     *
+     * @param string $class
+     * @param array  $metadata
+     */
+    protected function saveMetadata($class, $metadata)
+    {
+        if (!$this->trackMetadata) {
```

Comment:
disableFlag is set at-call, though, not a property on an object. I think these are separate types of use cases.